### PR TITLE
✨ Add class i-amphtml-story-interactive-active when current page is active to interactives

### PR DESCRIPTION
**Pull request by mszylkowski**
Subscribes to CURRENT_PAGE_ID changes and when the current page is active, it adds the class `i-amphtml-story-interactive-active` to the rootEl_.

This is the first step to implement the hint animations on pre-select, and other animations on the results pages that will trigger when the user lands on the page of the component. There's also a CSS variable `i-amphtml-story-interactive-animation-delay` that changes 0/350ms depending on the UI type.

Also: ordered properties alphabetically and called updateOptionPercentages from updateComponentToPostSelectionState

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-interactive.js [hunk 1]
```diff
@@ -381,6 +406,29 @@ export class AmpStoryInteractive extends AMP.BaseElement {
...
+                this.rootEl_.classList.add(
+                  'i-amphtml-story-interactive-active'
+                );
+              });
+            },
+            this.storeService_.get(StateProperty.UI_STATE) ===
+              UIType.DESKTOP_PANELS
+              ? 350
+              : 0
+          );
```
Comment (processprocess) [inline extensions/amp-story/1.0/amp-story-interactive.js:423]: What's the benefit of setting a delay here rather than using css transition delays?
We could use `i-amphtml-story-desktop-panels` in our stylesheets to detect if `amp-story` is in desktop mode.

Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story-interactive.js:423]: Good question, there's no definitive answer. In your codepen you were using transition delays to offset the animations between the different options, which would be harder to use: `calc(0.2s + var(--i-amphtml-interactive-animation-delay)))` vs just `0.2s`. Same thing would apply for any animation delays. And transitions that won't need a delay (eg: the results page), you wouldn't need to get the variable and set it to the animation-delay, just add the animations/transitions.

Comment (processprocess) [reply extensions/amp-story/1.0/amp-story-interactive.js:420]: I like the idea of having all timing logic in one place. 
However I see your point that it's verbose CSS. 

I would default to keeping concerns separated. 
Since this timing has to do with style, it feels logical to let CSS control it.

File: extensions/amp-story/1.0/amp-story-interactive.js [hunk 2]
```diff
@@ -193,6 +200,18 @@ export class AmpStoryInteractive extends AMP.BaseElement {
...
+   * @private
+   * @return {string} the page id
+   */
+  getPageId_() {
+    if (this.pageId_ == null) {
+      this.pageId_ = closest(dev().assertElement(this.element), (el) => {
+        return el.tagName.toLowerCase() === 'amp-story-page';
+      }).getAttribute('id');
+    }
+  }
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-interactive.js:213]: Did you forget to return the `pageId_`?

File: extensions/amp-story/1.0/amp-story-interactive.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-interactive.js:420]: +1 to handle the delay from the CSS.
`i-amphtml-story-interactive-active` is true even as the desktop page is transitioning (videos are playing etc)

File: extensions/amp-story/1.0/amp-story-interactive.js [hunk 3]
```diff
@@ -698,6 +728,7 @@ export class AmpStoryInteractive extends AMP.BaseElement {
 
     if (this.optionsData_) {
       this.rootEl_.classList.add('i-amphtml-story-interactive-has-data');
+      this.updateOptionPercentages_(this.optionsData_);
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-interactive.js:731]: Why this change?

Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story-interactive.js:731]: Saw that everywhere that updateToPostSelectionState was called, also updateOptionPercentages was being called. That's because updating the option percentages was part of the post-selection state for components, so I just moved the calls to this method to simplify the code.

File: extensions/amp-story/1.0/amp-story-interactive.js [hunk 4]
```diff
@@ -381,6 +398,19 @@ export class AmpStoryInteractive extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
+    // Check if the component page is active, and add class.
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      (currPageId) => {
+        if (currPageId === this.getPageId_()) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-interactive.js:405]: You'd also need to remove the active class when navigating away

File: extensions/amp-story/1.0/amp-story-interactive.js (hunk 1 above)
Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story-interactive.js]: Added commit that removes the delay here and adds a CSS variable for it

File: extensions/amp-story/1.0/amp-story-interactive.js (hunk 4 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-interactive.js:405]: 👀

Comment (mszylkowski) [reply extensions/amp-story/1.0/amp-story-interactive.js:405]: Sounds good for removing the active class. Meant anything else with the 👀?

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-interactive.js:408]: Nope I wanted to make sure you saw it as I sent it while you were pushing commits.

Can you remove the class in the `mutateElement` as well? You can use `classList.toggle(className, bool)`
